service/bulk_create.go

```diff
@@ -88,14 +88,26 @@ func BulkAssembly(req m.BulkCreateRequest, tenant *m.Tenant, user *m.User) (*m.B
 			}
 
 			if strings.ToLower(output.Authentications[i].ResourceType) == "application" {
-				output.ApplicationAuthentications = append(output.ApplicationAuthentications, m.ApplicationAuthentication{
+				applicationAuthentication := m.ApplicationAuthentication{
 					// TODO: After vault migration.
 					// VaultPath:         output.Authentications[i].Path(),
 					ApplicationID:    output.Authentications[i].ResourceID,
 					AuthenticationID: output.Authentications[i].DbID,
 					TenantID:         tenant.Id,
 					Tenant:           *tenant,
-				})
+				}
+
+				applicationFromAuthentication := &m.Application{ID: output.Authentications[i].ResourceID}
+				tx.Debug().Preload("ApplicationType").Find(&applicationFromAuthentication)
+
+				applicationType := &m.ApplicationType{Id: applicationFromAuthentication.ApplicationTypeID}
+				if tx.Debug().First(&applicationType).Error == nil {
+					if userResource.OwnershipPresentForApplication(applicationType.Name) {
+						applicationAuthentication.UserID = &userResource.User.Id
+					}
```

Comment:
I would suggest not using structs along with Gorm's `Find` functions to avoid issues like [RHCLOUD-20441](https://issues.redhat.com/browse/RHCLOUD-20441). The TL;DR is:

If the `out.put.Authentications[i].ResourceID` has a default value for any reason, Gorm will assume that it is safe not to include the `where application_id = 0` condition to the query, which will result in a `SELECT * FROM application`, and I guess that the `Find` method will pick up one of the results randomly.

So I'd suggest:

```diff
- applicationFromAuthentication := &m.Application{ID: output.Authentications[i].ResourceID}
+ var applicationFromAuthentication &m.Application
- tx.Debug().Preload("ApplicationType").Find(&applicationFromAuthentication)
+ tx.Debug().Preload("ApplicationType").Where("application_id = ?", output.Authentications[i].ResourceID).Find(&applicationFromAuthentication)
```

Reply:
That's is right and it is also for usages of gorm Find.

I change code little bit and I avoided first query because we have already fetched  in `output.Applications`


thanks!